query/query/query.py

```diff
@@ -229,6 +229,13 @@ async def on_cleanup(app):
     await asyncio.gather(*(t for t in asyncio.all_tasks() if t is not asyncio.current_task()))
 
 
+async def on_shutdown(app):
+    remaining_tasks = [t for t in asyncio.all_tasks() if t is not asyncio.current_task()]
+    log.info(f"On shutdown request received, with {len(remaining_tasks)} remaining tasks")
+    await asyncio.gather(*remaining_tasks)
```

Comment:
Is there an advantage of using `gather` over `wait` here?
(Also see https://stackoverflow.com/questions/42231161/asyncio-gather-vs-asyncio-wait).

Reply:
There's a disadvantage that there's an extra overhead for ensuring the tasks are ordered. As we don't care about that, I've changed it to `asyncio.wait`.